src/marqo/core/search/hybrid_search.py

```diff
@@ -0,0 +1,212 @@
+from typing import List, Optional, Union, Iterable, Dict
+
+from marqo.api import exceptions as api_exceptions
+from marqo.api import exceptions as errors
+# We depend on _httprequests.py for now, but this may be replaced in the future, as
+# _httprequests.py is designed for the client
+from marqo.config import Config
+from marqo.core import constants
+from marqo.core import exceptions as core_exceptions
+from marqo.core.models.hybrid_parameters import HybridParameters
+from marqo.core.models.marqo_index import UnstructuredMarqoIndex
+from marqo.core.models.marqo_query import MarqoHybridQuery
+from marqo.core.vespa_index import for_marqo_index as vespa_index_factory
+from marqo.core.structured_vespa_index.common import RANK_PROFILE_HYBRID_CUSTOM_SEARCHER
+from marqo.tensor_search import index_meta_cache
+from marqo.tensor_search import utils
+from marqo.tensor_search.enums import (
+    SearchMethod
+)
+from marqo.tensor_search.models.api_models import BulkSearchQueryEntity, ScoreModifierLists, CustomVectorQuery
+from marqo.tensor_search.models.private_models import ModelAuth
+from marqo.tensor_search.models.search import Qidx, SearchContext, SearchContextTensor
+from marqo.tensor_search.telemetry import RequestMetricsStore
+from marqo.tensor_search.tensor_search import run_vectorise_pipeline, gather_documents_from_response, logger
+from marqo.vespa.exceptions import VespaStatusError
+
+
+class HybridSearch:
+    def search(
+            self, config: Config, index_name: str, query: Optional[Union[str, CustomVectorQuery]],
+            result_count: int = 5,
+            offset: int = 0, ef_search: Optional[int] = None, approximate: bool = True,
+            searchable_attributes: Iterable[str] = None, filter_string: str = None, device: str = None,
+            attributes_to_retrieve: Optional[List[str]] = None, boost: Optional[Dict] = None,
+            image_download_headers: Optional[Dict] = None, context: Optional[SearchContext] = None,
+            score_modifiers: Optional[ScoreModifierLists] = None, model_auth: Optional[ModelAuth] = None,
+            highlights: bool = False, text_query_prefix: Optional[str] = None,
+            hybrid_parameters: HybridParameters = None) -> Dict:
+        """
+
+            Args:
```

Comment:
is this copied from another method? We'll need to remove some of them, and add documents of hybrid_parameters

Reply:
done